MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -150,26 +149,38 @@ - (NSDictionary *) convertSpotifyTracks:(NSArray *)tracks{
 }
 -(void) saveSpotifyData:(NSDictionary *) spotifyData{
     //Saves all Spotify Data such as there top genres,tracks,artist,album,and the artist images to the Parse Database
+    PFObject *music = [PFObject objectWithClassName:@"Music"];
     PFUser *current = [PFUser currentUser];
-    current[@"genres"] = [NSArray arrayWithArray:spotifyData[@"genres"]];
-    current[@"tracks"] = [NSArray arrayWithArray:spotifyData[@"tracks"]];
-    current[@"artists"] = [NSArray arrayWithArray:spotifyData[@"artists"]];
-    current[@"albums"] = [NSArray arrayWithArray:spotifyData[@"albums"]];
-    current[@"images"] = [NSArray arrayWithArray:spotifyData[@"images"]];
-    [PFUser.currentUser saveInBackground];
+    PFQuery *query = [PFQuery queryWithClassName:@"Music"];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *results, NSError *error) {
+        if (![[results valueForKey:@"userId"]containsObject:current.objectId]){
+            //Checks if User has already been added to database before adding
+            music[@"userId"] = current.objectId;
+            music[@"genres"] = [NSArray arrayWithArray:spotifyData[@"genres"]];
+            music[@"tracks"] = [NSArray arrayWithArray:spotifyData[@"tracks"]];
+            music[@"artists"] = [NSArray arrayWithArray:spotifyData[@"artists"]];
+            music[@"albums"] = [NSArray arrayWithArray:spotifyData[@"albums"]];
+            music[@"images"] = [NSArray arrayWithArray:spotifyData[@"images"]];
+            [music saveEventually];
+        }
+    }];
 }
+
 -(void) saveSpotifyUserData:(NSDictionary *) spotifyData{
     // Saves name and Profile Picture to User
     [self getSpotifyData:@"https://api.spotify.com/v1/me" completion:^(NSDictionary * userDict, NSError * error) {
         if (!error){
+            PFUser *current = [PFUser currentUser];
             NSString *name = userDict[@"display_name"];
             NSArray *image= userDict[@"images"];
             NSDictionary *imageDict = [image objectAtIndex:0];
-            NSString * url =[NSString stringWithFormat:imageDict[@"url"], nil];
-            [User user].profilePicture = url;
-            [User user].name = [NSString stringWithFormat: name, nil];
-                }
-        }];
+            if (!imageDict == nil){
+                [User user].name = [NSString stringWithFormat: name, nil];
```

Comment:
Made sure the imageDict isn't nil before declaring objects. In later iterations, there will be a default image that will be placed if there is no image for the profile